### ISSUE TITLE
ocean-ice velocities should be exchanged on C-grid when running w/ CICE6 C-grid

### ISSUE THREAD

**Issue by DeniseWorthen** [ufs-community/ufs-weather-model#1555]
## Description

CICE6 runs on the B-grid unless C-grid is specified. Velocities are located on the corners. MOM6 runs on the C-grid and velocities are located on the E and N faces. For exchange of velocity components between CICE6 and MOM6 in B+C grid case, the velocities are first averaged to the center grid point before being exchanged.

When CICE6 runs on the C-grid, no averaging of velocities is required and the velocities should be exchanged on the native E/N grid points. 

## Solution

When CICE6 is run on the C-grid, velocity components should be exchanged on the native grid points.

**Comment by DeniseWorthen** [comment]
@NeilBarton-NOAA This is the UWM issue for c-grid compatability.

**Comment by gspetro-NOAA** [comment]
@DeniseWorthen Is this issue still planned/desired? Or has it been resolved?

**Comment by DeniseWorthen** [comment]
@gspetro-NOAA Yes, @NickSzapiro-NOAA and I plan to work on this in the near future.

**Comment by DeniseWorthen** [comment]
I've been looking into this issue, a few points I've noted.

1) Even if the OCN u,v are exchanged on the C-grid, we still require the OCN u,v fields on the A-grid to be passed out by the MOM6 cap. The A-grid fields are used by both the ATM (when ``use_oceanuv=.true``) and by the WAV model. In the MOM6 cap, the ocean velocities are available only from the ``ocean_public`` derived type.  

   In ``mom_ocean_model_nuopc.F90``, the ocean_public surface velocities are filled w/ the correct stagger values depending    on the ``OCEAN_SURFACE_STAGGER`` setting from MOM_input. I believe that means that when we set the desired stagger to be ``C``, we will then need to calculate the ``A`` grid values in the cap, since we still require these values to be exported. So, we'll need something like the current ``So_u,So_v`` which will be filled with A-grid velocities and new coupling fields ``So_uc,So_vc`` which will come from ocean_public on the C-grid.

2. It's not clear to me exactly what the ``grid_ocn`` setting needs to be on the CICE side, if we supply the ocean velocities on the ``C`` grid.  We want the dynamics to run on the C-grid (``grid_ice=C``), but we want the fluxes to still be calculated on the A-grid; that includes the ocean-ice stresses that need to be exported by the cice cap.

**Comment by sanAkel** [comment]
I'm positive of this that the ocean needs "total" stress, i.e., 
Stress from winds (atm) +
Stress from sea ice

Please just make sure of the ⬆️

Thanks!

**Comment by DeniseWorthen** [comment]
I'm no expert at the ATM source code, but it looks consistent that the surface currents need to be on the center locations to be used in the stress calculation (https://github.com/NCAR/ccpp-physics/blob/52a9db3a893e2d3835b57bb0d6f262cb7c971c30/physics/PBL/SATMEDMF/satmedmfvdifq.F#L3054-L3059):

```
      if (use_oceanuv) then
        do i = 1,im
          spd1_m=sqrt( (u1(i,1)-usfco(i))**2+(v1(i,1)-vsfco(i))**2 )
          dusfc(i) = -1.*rho_a(i)*stress(i)*(u1(i,1)-usfco(i))/spd1_m
          dvsfc(i) = -1.*rho_a(i)*stress(i)*(v1(i,1)-vsfco(i))/spd1_m
        enddo
      else
```
 
where u1,v1 are the bottom velocities in the ATM on the mass grid (centers).

**Comment by sanAkel** [comment]
Great!

Please verify that the stress 

Sent to ocean: includes atm +
Ice

**Comment by DeniseWorthen** [comment]
@sanAkel The merging (summing) of the ATM and ICE stresses happens in the mediator where the ATM ``taux,tauy`` (on the A-grid) is combined with the ICE-OCN stresses (computed by CICE, on the A-grid). The ocean gets the ice-ocean (weighted by ifrac) + atm-ocean (weighted by ofrac). See

https://github.com/NOAA-EMC/CMEPS/blob/27e3961d9bb323fc5ff16a532947caddb515eb95/mediator/esmFldsExchange_ufs_mod.F90#L475

**Comment by sanAkel** [comment]
Thanks @DeniseWorthen

**Comment by DeniseWorthen** [comment]
I have a feature branch in UWM to test this concept, using modifications to the CICE and MOM6 caps and CMEPS. In order to evaluate whether the "C-grid exchange" is beneficial, I created a DATM configuration using the 1deg MOM6/CICE. Since I had no DATM forcing which covers a Arctic summer season, I created it by running the current ``cpld_control_nowave_noaero`` RT and created the DATM forcing from the gaussian output for 2021-05-01 thru 2021-08-29. I then used that DATM forcing with three configurations

| MOM6 export grid | CICE grid_ocn | CICE grid_ice |           |
| -------------------| -------------- | -------------- | ----- |
| A | A | B | AB case; current default; MOM exports all variables on A grid, CICE dyn runs on B grid |
| A | A | C | AC case; current capabilty for C-grid dynamics; MOM exports all variables on A grid, CICE dyn runs on C grid |
| C | C | C | CC case; new capability for C-grid dynamics; MOM exports C grid dynamic variables, CICE dyn runs on C grid |

I plotted the icespd variables from the CICE history files. For the AB case, these speed variables are on the B (corners); for the AC and CC cases, I picked either the icespdN_h or icespdE_h variables to plot, depending on whether the transect was more N-S or more E-W (the results were similar regardless of which face I picked).

<img width="1148" height="1356" alt="Image" src="https://github.com/user-attachments/assets/103c83ea-cdd6-48ba-ac57-29bfa55b426c" />


Assuming my AI friend made the plots correctly, this test does seem to indicate that exporting C-grid velocities and slopes from MOM6 produces a small increase in ice flow through narrow channels compared to sending A-grid velocities and slopes. Either is better than using the B-grid dynamics in CICE.

My UWM branch is https://github.com/DeniseWorthen/ufs-weather-model/tree/feature/CgridFrOcn

**Comment by sanAkel** [comment]
These are very good illustrations @DeniseWorthen to show that the mechanics seem to be working as intended.
The contrast between "A:B" (blue) and A:C, C:C is clear, your table states what's what!

1. I have a question reg CICE `grid_ocn` and `grid_ice` - will reach offline.
2. Perhaps @dabail10, @zhaobin74 can comment more ... I do not know details of CICE dynamics.

**Comment by dabail10** [comment]
Is this only the ice-ocean velocities or does this include the ice-ocean stresses? The ice-ocean stresses are merged by fraction with the air-ocean stresses and sent to the ocean. I see a comment above about this.

**Comment by sanAkel** [comment]
> Is this only the ice-ocean velocities or does this include the ice-ocean stresses? The ice-ocean stresses are merged by fraction with the air-ocean stresses and sent to the ocean. I see a comment above about this.

@dabail10 AFIK it's about velocities- right @DeniseWorthen?

**Comment by DeniseWorthen** [comment]
Yes, this is only sending the velocities and the surface slopes on the C-grid from OCN->ICE.

**Comment by dabail10** [comment]
> Yes, this is only sending the velocities and the surface slopes on the C-grid from OCN->ICE.

Ah, ok. Does this not cause problems with MOM receiving the taux and tauy on the A-grid?

**Comment by DeniseWorthen** [comment]
This does not change the forcing field stagger that MOM6 receives. That is set by the MOM_input value of ``WIND_STAGGER``. The MOM6 forcing stagger still needs to be on the A-grid because it represents a merge of ATM and ICE forcing. 

As far as I understand C-grid in CICE, the thermo calculations are always done on the A-grid, regardless of what ``grid_ocn`` or ``grid_ice`` is set at.

**Comment by sanAkel** [comment]
> This does not change the forcing field stagger that MOM6 receives. That is set by the MOM_input value of `WIND_STAGGER`. The MOM6 forcing stagger still needs to be on the A-grid because it represents a merge of ATM and ICE forcing.
> 

@dabail10 Please see the  [WIND_STAGGER](https://github.com/NOAA-GFDL/MOM6-examples/blob/695294882794f389eb7bf7cc965ec9896e78fbc6/ice_ocean_SIS2/OM4_025/MOM_parameter_doc.all#L2645-L2647) settings in MOM (clearly it can be set to `A`- `C`).

> As far as I understand C-grid in CICE, the thermo calculations are always done on the A-grid, regardless of what `grid_ocn` or `grid_ice` is set at.
